src/controllers/ClassInfoController.java

```diff
@@ -42,6 +47,26 @@ public void initialize(URL location, ResourceBundle resources) {
 		this.classes = SubjectDatabaseController.getAllClasses();
 		this.classIndex = 0;
 	}
+  
+	public void btnDeleteSubjectAction(){
+		if(classes.size() > 0)
+			SubjectDatabaseController.deleteSubject(classes.get(classIndex).getSubjectId());
+		
```

Comment:
You should call the parent.refreshData() method to let know the parent that the info has changed

Reply:
Roger roger, all done :robot: